core/templates/dev/head/services/UserService.ts

```diff
@@ -17,48 +17,56 @@
  */
 
 oppia.factory('UserService', [
-  '$http', '$q', '$window', 'UrlInterpolationService', 'UserInfoObjectFactory',
-  'DEFAULT_PROFILE_IMAGE_PATH',
-  function($http, $q, $window, UrlInterpolationService, UserInfoObjectFactory,
-      DEFAULT_PROFILE_IMAGE_PATH) {
+  '$http', '$q', '$window', 'UrlInterpolationService',
+  'UserInfoObjectFactory', 'DEFAULT_PROFILE_IMAGE_PATH',
+  function($http, $q, $window, UrlInterpolationService,
+      UserInfoObjectFactory, DEFAULT_PROFILE_IMAGE_PATH) {
     var PREFERENCES_DATA_URL = '/preferenceshandler/data';
 
     var userInfo = null;
 
     var getUserInfoAsync = function() {
-      if (GLOBALS.userIsLoggedIn) {
-        if (userInfo) {
-          return $q.resolve(userInfo);
-        }
-        return $http.get(
-          '/userinfohandler'
-        ).then(function(response) {
-          userInfo = UserInfoObjectFactory.createFromBackendDict(response.data);
-          return userInfo;
-        });
-      } else {
-        return $q.resolve(UserInfoObjectFactory.createDefault());
+      var currentUrl = $window.location.href;
+      if (currentUrl.includes('signup')) {
+        userInfo = UserInfoObjectFactory.createDefault();
+        return $q.resolve(userInfo);
+      }
+      if (userInfo) {
+        return $q.resolve(userInfo);
       }
+      return $http.get(
+        '/userinfohandler'
+      ).then(function(response) {
+        if (response.data.user_is_logged_in) {
+          userInfo = (
+            UserInfoObjectFactory.createFromBackendDict(response.data));
+        } else {
+          userInfo = UserInfoObjectFactory.createDefault();
```

Comment:
I'm not sure but the default object handles logged out user case?

Reply:
I think it handles logged out user case.
UserInfoObjectFactory.createDefault(); will create UserInfo object with 
'is_moderator': False,
'is_admin': False,
'is_super_admin': False,
'is_topic_manager': False,
'can_create_collections': False,
'username': None,
'preferred_site_language_code': None
 'user_is_logged_in': False